spec/bolt/inventory/inventory2_spec.rb

```diff
@@ -682,6 +687,329 @@ def common_data(transport)
         end
       end
     end
+
+    describe 'get_target' do
```

Comment:
You could have
```
let(:inventory) { Bolt::Inventory::Inventory2.new(data) }
```
under the `describe`, then just set data where it's special (including `{}` for the empty inventory).

Reply:
The other contexts are not nested though? Again, this is replicated from above. Personally I find the memo definitions being in "close" proximity to where they are used can be really helpful. Though I see how it is kind of a mess in this file. If you feel strongly about it I can try a bigger refactor.